engine/launcher/src/main/scala/org/enso/launcher/archive/ArchiveIterator.scala

```diff
@@ -0,0 +1,45 @@
+package org.enso.launcher.archive
+
+import org.apache.commons.compress.archivers.{ArchiveEntry, ArchiveInputStream}
+
+/**
+  * Wraps an [[ArchiveInputStream]] to get an [[Iterator]] which produces
+  * non-null archive entries.
+  */
+private[archive] case class ArchiveIterator(
```

Comment:
Because it's internal to the implementation of the `Archive` utilities. It's not needed outside of it, and I tend to prefer private by default, but can change that.